### PR TITLE
fix: remove finngen prefix from credible set

### PR DESCRIPTION
**Pull request by project-defiant**
## ✨ Context

@xyg123 have found a bug in the finngen_r11 `study_index` dataset. The `studyId` did not have the `FINNGEN_R11_` prefix. that was found in finemapping results.

* study_index
```
+------------------------+
|studyId                 |
+------------------------+
|I9_AVBLOCK              |
|I9_ABAORTANEUR          |
|O15_ANTENAT_ABNORM      |
|Q17_APVR                |
|H8_OSSICABN             |
|O15_LABOUR_ABNORM_FORCES|
|K11_ABRASION            |
|N14_BARTHOABSCESS       |
````

* finngen_credible_set
```
+----------------------------------------+
|studyId                                 |
+----------------------------------------+
|FINNGEN_R11_ALCOHOLACUTE10              |
|FINNGEN_R11_ALLERG_ASTHMA               |
|FINNGEN_R11_ALLERG_ASTHMA               |
|FINNGEN_R11_ANTIDEPRESSANTS             |
|FINNGEN_R11_ASTHMA_EOSINOPHIL_SUGG      |
```

## 🛠 What does this PR implement

This PR fixes the issue with missing prefix and adds test to ensure the default prefix is found when reading from source.

## 🚦 Before submitting

- [x] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [ ] Did you make sure the branch is up-to-date with the `dev` branch?
- [ ] Did you write any new necessary tests?
- [ ] Did you make sure the changes pass local tests (`make test`)?
- [ ] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

**Review by ireneisdoomed — COMMENTED** [review]
As part of the work in the study index, it was decided that we don't want this prefix in the study IDs

**Comment by project-defiant** [comment]
@ireneisdoomed in that context I will drop it from the finemapping and revert this change. Thank you!

**Comment by project-defiant** [comment]
@ireneisdoomed ready for round 2

File: src/gentropy/datasource/finngen/study_index.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def join_efo_mapping(
         )
         if not finngen_release_prefix_match:
             raise ValueError(
-                f"Invalid FinnGen release prefix: {finngen_release_prefix}, use the format FINNGEN_R*_"
+                f"Invalid FinnGen release prefix: {finngen_release_prefix}, use the format FINNGEN_R*"
```
Comment (ireneisdoomed) [inline src/gentropy/datasource/finngen/study_index.py:58]: Probably the prefix doesnt make sense here either. What do you think?

**Review by ireneisdoomed — APPROVED** [review]
Ty! My comment is not about this specific PR, just a comment for a future iteration.

Please change the PR title

**Comment by project-defiant** [comment]
Thank you. Just realised is opposite now

**Pull request merged**